podpac/core/coordinates/coordinates1d.py

```diff
@@ -350,19 +350,31 @@ def select(self, bounds, return_indices=False, outer=False):
             index or slice for the selected coordinates (only if return_indices=True)
         """
 
+        # empty case
+        if self.dtype is None:
+            return self._select_empty(return_indices)
+
         if isinstance(bounds, dict):
             bounds = bounds.get(self.name)
             if bounds is None:
                 return self._select_full(return_indices)
 
         bounds = make_coord_value(bounds[0]), make_coord_value(bounds[1])
+
+        # check type
+        if not isinstance(bounds[0], self.dtype):
+            raise TypeError(
+                "Input bounds do match the coordinates dtype (%s != %s)" % (type(self.bounds[0]), self.dtype)
+            )
+        if not isinstance(bounds[1], self.dtype):
+            raise TypeError(
+                "Input bounds do match the coordinates dtype (%s != %s)" % (type(self.bounds[1]), self.dtype)
+            )
+
         my_bounds = self.area_bounds.copy()

```

Comment:
Will this throw errors when self.dtype == float and bounds == int?

Reply:
I don't think so, because `make_coordinate_value` converts ints to floats.

Reply:
Awesome. I like this check -- it shows that you have an intimate knowledge of this class. :)